### PR TITLE
Add remove functions implementation

### PR DESCRIPTION
**Pull request by Hombre2014**
### Phase 2, create and remove books
- [x] Configuring the Redux Store
- [x] Write book's actions and reducer
- [x] In the React component responsible for adding new books set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step). Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action.
- [x] In the React component responsible for removing books - implement that event by dispatching a corresponding action.
- [x] Use the redux-logger to check if your application is working correctly - Open your browser's console and look for the logger messages. Upon every action dispatch, the logger will print:
  - [x] prev state - the state of the whole Redux store before the current action was dispatched.
  - [x] action - the action object that is being dispatched.
  - [x] next state - the state of the Redux store after the current action was dispatched.

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { createStore, combineReducers, applyMiddleware } from 'redux';
+import logger from 'redux-logger';
+import booksReducer from './books/books';
+// import { initialState } from './books/books';
```
Comment (ZeenatLawal) [inline src/redux/configureStore.js]: - [x] Kindly remove commented out codes to follow best practices.

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+const reducer = combineReducers({
+  booksReducer,
+});
+
+const store = createStore(
+  reducer,
+  // (state, action) => reducer(state, action), initialState,
```
Comment (ZeenatLawal) [inline src/redux/configureStore.js]: - [x] Kindly remove commented out codes to follow best practices.

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (ZeenatLawal) [inline src/redux/books/books.js]: - [x] Action types should be written using ducks pattern as stated in the requirements. The action types should be written in a format like this `const ADD_BOOK = 'appname/reducer/ACTION_TYPE';`. Kindly modify this.

File: src/redux/books/books.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  type: REMOVE_BOOK,
+  payload,
+});
+
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.payload];
+    case REMOVE_BOOK:
+      return [...state.filter((book) => book.id !== action.payload)];
```
Comment (ZeenatLawal) [inline src/redux/books/books.js]: - [x] The REMOVE_BOOK action type should return a new array as stated in the requirements. I would suggest applying the filter directly without spreading it in another array, you could write it like this `return state.filter((book) => book.id !== action.payload);`

**Review by ZeenatLawal — CHANGES_REQUESTED** [review]
Hi @Hombre2014, 

Good job so far!:clap: :clap:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎊 🎊 
- Git flow is followed. ✔️
- PR has a title and summary. ✔️


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hombre2014** [comment]
Thanks for your comment and recommendations. All the required changes have been implemented.

**Review by dasileker — APPROVED** [review]
## **Hi @Hombre2014 ,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**